dev/com.ibm.ws.jaxws.2.2.webcontainer_fat_extended/fat/src/com/ibm/ws/jaxws/fat/HandlerChainTest.java

```diff
@@ -48,6 +51,9 @@ public class HandlerChainTest {
     @Server("XmlOverrideHandlerChainTestServer")
     public static LibertyServer xmlServer;
 
+    @Server("AddedElementHandlerChainTestServer")
```

Comment:
Is there a reason to add an additional server for this test? It seems like the new tests applications could be added to the xmlServer instead. The reason I ask is that every server we add to a test suite, means adding an additional hit for server start and stop.

Reply:
Whenever we use the same server for both tests, with Semeru 17 java core issues are observed, with 2 others that I tried, Java 2 security issue happened on java.util.PropertyPermission even though we provide all permissions. I think we have a problem stopping the server cleanly for these cases. We observed this behavior on few more test cases before. Although adding a new server is resource consuming, it seemed like a cleaner solution.

Reply:
Interesting. I would argue that adding more than one sever to a test suite actually does the opposite, and makes the solution more complex. It doesn't need to be done with this pull request, but we should take a `TO-DO` to split this test suite in to multiple test classes. If the tests don't require server to server communication, then best practice is to have one Liberty server per test suite, which I know that a lot of our test suites don't do already, but going forward we need to put it into practice by either adding new test suites for new tests that require a unique server, or by adding new test applications to the existing server in the suite.

Can you add a comment to that effect?  
```suggestion
    // TODO: Refactor to remove multiple servers from what should be a single server test suite. 
    @Server("AddedElementHandlerChainTestServer")
```

Reply:
I meant by cleaner, less problematic. Otherwise, I agree. It's better to have one server unless it's not %100 required per test class. It's cleaner by design :)